code/datums/datumvars.dm

```diff
@@ -715,7 +715,10 @@
 		log_admin("[key_name(usr)] has offered control of ([key_name(M)]) to ghosts.")
 		var/minhours = input(usr, "Minimum hours required to play [M]?", "Set Min Hrs", 10) as num
 		message_admins("[key_name_admin(usr)] has offered control of ([key_name_admin(M)]) to ghosts with [minhours] hrs playtime")
-		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [M.real_name]?", poll_time = 100, min_hours = minhours)
+		var/mname = M.name
+		if(M.real_name)
+			mname = M.real_name
+		var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as [mname]?", poll_time = 100, min_hours = minhours)

```

Comment:
Consider the ternary operator as an alternative that doesn't require an extra var and an awkward if in the middle of this block.

ie; 
`var/list/mob/dead/observer/candidates` = pollCandidates("Do you want to play as [M.real_name ? "M.real_name" : "M.name"]?", poll_time = 100, min_hours = minhours)`